sys/bhp/Kconfig

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2022 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+menu "Bottom Half Processor"
+
+config MODULE_BHP
+    bool
+
+config MODULE_BHP_EVENT
+    bool "Enable event based Bottom Half Processor implementation"
+    depends on MODULE_EVENT
+    select MODULE_BHP
```

Comment:
Seems a bit redundant to define both?
@leandrolanzieri might know better

Reply:
I think it is relatively low-risk here to `select MODULE_EVENT`, if that's what you mean

Reply:
Hmmm now that I see it, probably will need to change the modelling a bit.
My idea was to use `MODULE_BHP` as a feature request that the driver can select.
Then, another module (e.g `gnrc_netif`) can select `MODULE_BHP_EVENT` **if** `MODULE_BHP` is present. That way we don't pull BHP code when devices don't require it

Reply:
I just had an offline discussion with @leandrolanzieri and we agreed on leaving `MODULE_BHP` to compile `sys/bhp`. This is also used to indicate in Make that a driver requires a Bottom Half Processor, **but** it doesn't force the user to bring any of the `bhp_*` modules.
For Kconfig I simply added a `HAVE_BHP_IRQ_HANDLER` feature to the driver that indicates there's an IRQ handler expected to be offloaded.